Eastern-Research-Group/mywaterway
# Feature/572 individual characteristics modal

## Related Issues:
* [HMW-572](https://jira.epa.gov/browse/HMW-572)

## Main Changes:
* Added a modal and trigger to the _Past Water Conditions_ list content, which shows the individual characteristics that make up each top-level characteristic group, along with their measurement counts.
  * The counts do not always align, i.e. a group's "Number of Measurements" may not equal the count calculated by adding each characteristic's "Number of Measurements". This is due to the fact that the WQP domain values service is missing a number of characteristics. This service was used to create the `characteristicsByGroup` mapping file (it is only provided in a ZIP CSV file, so parsing was done ahead of time; however, I have confirmed that the file's content is up-to-date).
* Moved the `characteristicsByGroup` file to the S3 bucket.
* Changed more occurrences of _CyAN_ to _Blue-Green Algae_ internally. A type mismatch was preventing content from rendering in the lists.
* Addressed a small map-related memory leak.
* Adjusted the `Modal` component to scroll when overflowing.

## Steps To Test:
1. Go to http://localhost:3000/community/dc/monitoring.
2. In the Past Water Conditions section, open an accordion item.
3. Click the information icon in the item's toggle table, next to a characteristic group.
4. Verify a modal appears with a list of individual characteristics and counts.
5. On the map, select the location shown below: 
![overflow](https://github.com/Eastern-Research-Group/mywaterway/assets/104089016/06ba128c-c646-4416-9c92-3e6b6e1f787b)
6. Open the modal for the **Pesticides** group, and confirm the overflow is scrollable.
7. Repeat in the Overview tab.
8. Go to the Past Water Conditions section of http://localhost:3000/community/lake%20okeechobee/monitoring, then sort the list by **Monitoring Measurements**.
9. Open the **Nutrients** modal, and take note of the counts.
10. Adjust the date slider, then repeat step 8. The counts should differ.
11. Open the corresponding map popup (by clicking the location on the map, not by the _View on Map_ button, although the button can be used to find the location), and check the Nutrients characteristics counts again. They should match the counts found in the list.
12. Verify the CyAN content is loading correctly in the Current Water Conditions tab.
